### PR TITLE
Add animation preset squash and stretch and some namui chagnes

### PR DESCRIPTION
**Pull request by namse**
# What I did
- React-ished dropdown
- Add animation preset 'squash and stretch'
- Removed graph editor
- Use matrix instead of xywh for animation image 
- and some small patches

### CODE REVIEW

File: namui-prebuilt/src/dropdown/mod.rs [hunk 1]
```diff
@@ -38,55 +38,53 @@ where
...
+        || {
+            Box::new(Dropdown {
+                id: namui::random_id(),
+                is_opened: false,
+                list_view: list_view::ListView::new(),
+                mouse_over_item_id: None,
+            })
+        },
+        internal_props,
+    )
```
Comment (namse) [inline namui-prebuilt/src/dropdown/mod.rs:71]: Dropdown become reacted. it's not stable or comfortable to make some entity as reacted. I know it and will make improvement.

File: namui/src/namui/animation/keyframe.rs [hunk 2]
```diff
@@ -21,29 +21,47 @@ impl<T: Clone> KeyframePoint<T> {
...
+    fn interpolate(&self, next: &Self, context: &InterpolationContext<TKeyframeLine>) -> Self;
 }
 
-pub trait KeyframeValue {
-    fn interpolate(&self, next: &Self, ratio: f32) -> Self;
+pub struct InterpolationContext<'a, TKeyframeLine> {
+    pub line: &'a TKeyframeLine,
+    pub time_ratio: f32,
+    pub duration: Time,
 }
```
Comment (namse) [inline namui/src/namui/animation/keyframe.rs:32]: Animation Interpolation System little bit changed, more expandable.

File: namui/src/namui/common/types/time.rs [hunk 3]
```diff
@@ -5,18 +5,53 @@ use ordered_float::OrderedFloat;
...
+    Minute(f32),
     Hour(f32),
     Day(f32),
     Week(f32),
 }
 
+pub trait TimeExt {
+    fn ms(self) -> Time;
+    fn sec(self) -> Time;
+    fn minute(self) -> Time;
```
Comment (namse) [inline namui/src/namui/common/types/time.rs:17]: The reason why I change `min` to `minute` is becuase `1.min()` conflicts with math's min(imum) and min(ute).

**Comment by namse** [comment]
You can reject and suggest to me to divide changes about namui and editor, prebuilt in each pull requests.

**Review by bigfoodK — APPROVED** [review]

**Comment by bigfoodK** [comment]
> You can reject and suggest to me to divide changes about namui and editor, prebuilt in each pull requests.

It's not good idea, but it's not bad idea either.
I don't think there will be a big problem even if it is included in this pr.

**Pull request merged**